File: app/src/main/java/org/zeroxlab/momodict/widget/CardRowPresenter.java
package org.zeroxlab.momodict.widget;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.zeroxlab.momodict.R;
import org.zeroxlab.momodict.model.Card;

public class CardRowPresenter implements SelectorAdapter.Presenter<Card> {

    View.OnClickListener mListener;
    View.OnLongClickListener mLongListener;

    public CardRowPresenter(View.OnClickListener listener,
                            View.OnLongClickListener longListener) {
        mListener = listener;
        mLongListener = longListener;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        ViewGroup viewGroup = (ViewGroup) inflater.inflate(
                R.layout.list_item_card_row, parent, false);
        return new InnerViewHolder(viewGroup);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, Card item) {
        InnerViewHolder holder = (InnerViewHolder) viewHolder;
        holder.iText1.setText(item.wordStr);
        holder.itemView.setOnClickListener(view -> {
            view.setTag(item.wordStr);
            mListener.onClick(view);
        });
        holder.itemView.setOnLongClickListener(view -> {
            view.setTag(item.wordStr);
            mLongListener.onLongClick(view);
            return true;
        });
    }

    @Override
    public void onUnbindViewHolder(RecyclerView.ViewHolder viewHolder) {
    }

    class InnerViewHolder extends RecyclerView.ViewHolder {
        TextView iText1;
        TextView iText2;

        public InnerViewHolder(View view) {
            super(view);
            iText1 = (TextView) view.findViewById(R.id.text_1);
            iText2 = (TextView) view.findViewById(R.id.text_2);
        }
    }
}

File: app/src/main/java/org/zeroxlab/momodict/widget/CardRowPresenter.kt
package org.zeroxlab.momodict.widget

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import org.zeroxlab.momodict.R
import org.zeroxlab.momodict.model.Card

class CardRowPresenter(private val clickCallback: (v: View) -> Unit,
                       private val longClickCallback: (v: View) -> Unit) : SelectorAdapter.Presenter<Card> {

    override fun onCreateViewHolder(parent: ViewGroup): RecyclerView.ViewHolder {
        return LayoutInflater
                .from(parent.context)
                .inflate(R.layout.list_item_card_row, parent, false)
                .let { view -> view as ViewGroup }
                .let { viewGroup -> InnerViewHolder(viewGroup) }
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, item: Card) {
        val holder = viewHolder as InnerViewHolder

        holder.iText1.text = item.wordStr

        holder.itemView.setOnClickListener { view ->
            view.tag = item.wordStr
            clickCallback(view)
            true
        }

        holder.itemView.setOnLongClickListener { view ->
            view.tag = item.wordStr
            longClickCallback(view)
            true
        }
    }

    override fun onUnbindViewHolder(viewHolder: RecyclerView.ViewHolder) {}

    internal inner class InnerViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var iText1: TextView = view.findViewById<View>(R.id.text_1) as TextView
    }
}
